proxy/api/src/main.rs

```diff
@@ -20,46 +32,121 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
 
     let mut args = pico_args::Arguments::from_env();
     let args = Args {
-        reset_state: args.contains("--reset-state"),
         test: args.contains("--test"),
     };
 
-    if args.reset_state {
-        return Ok(reset_state()?);
-    }
-
     let proxy_path = config::proxy_path()?;
     let bin_dir = config::bin_dir()?;
     coco::git_helper::setup(&proxy_path, &bin_dir)?;
 
-    let temp_dir = tempfile::tempdir()?;
-    log::debug!(
-        "Temporary path being used for this run is: {:?}",
-        temp_dir.path()
-    );
+    let mut sighup = signal(SignalKind::hangup())?;
+    loop {
+        let rigging = rig(args).await?;
+        let (mut tx, rx) = mpsc::channel(1);
+        let runner = run(rigging, (tx.clone(), rx), args.test);
+
+        tokio::select! {
+            r = runner => match r {
+                // We've been shut down, ignore
+                Err(RunError::Peer(coco::peer::Error::Aborted(_))) | Ok(()) => {
+                    log::debug!("aborted")
+                },
+                // Actual error, abort the process
+                Err(e) => return Err(e.into()),
+            },
 
-    let paths_config = if args.test {
-        std::env::set_var("RAD_HOME", temp_dir.path());
-        coco::config::Paths::FromRoot(temp_dir.path().to_path_buf())
-    } else {
-        coco::config::Paths::default()
-    };
-    let paths = coco::Paths::try_from(paths_config)?;
+            Some(()) = sighup.recv() => {
+                log::info!("SIGHUP received, reloading...");
```

Comment:
To the SIG HUP you drop gitit?

Because with `--test`, everything is in a temp dir, this is equivalent to a reset. Without `--test`, it's equivalent to, well, restarting.